src/cli/commands/execute.rs

```diff
@@ -57,8 +63,39 @@ impl Display for ExecuteOutput {
 
 impl CommandExecution<ExecuteOutput> for ExecuteArgs {
 	fn exec(&self) -> Result<ExecuteOutput, String> {
-		let mut cairo_runner =
-			cairo_run(&self.program, "main", false, &HINT_EXECUTOR).map_err(|e| {
+		// Use cairo-compile binary in order to compile the .cairo file
+		let compiled_output: Output = Command::new("cairo-compile")
+			.args([&self.program])
+			.output()
+			.expect("cairo-compile command failed to start");
+
+		// Retrieve only the file name to create a clean compiled file name.
+		let program = self.program.display().to_string();
+		let filename: Vec<&str> = program.split('/').collect();
+		let filename = filename[filename.len() - 1].replace(".cairo", "");
+
+		// Build path to save the  compiled file
+		let dir = String::from("./test_cairo_contracts/compiled_contracts/");
+		let file_extension = String::from(".json");
+
+		let mut path = String::new();
+		path.push_str(&dir);
+		path.push_str(&filename);
+		path.push_str(&file_extension);
+
+		// Create a file to store command output inside a json file
+		let file = File::create(path.clone());
+		let _file = match file {
+			Ok(mut file) => file.write_all(&compiled_output.stdout),
+			Err(error) => {
+				panic!("Failed to store previous output {:?}", error)
+			},
+		};
+
+		// Run the main function of cairo contract
+		let path_compiled_file = Path::new(&path);
```

Comment:
Everything up until this should be put in a separate `compile` module.

Here is a proposition of architecture:
```
src/cli/command/
    list.rs
    execute/
        mod.rs
        tests.rs
        compile.rs
```